exchange/bitswap/wantmanager.go

```diff
@@ -168,11 +168,17 @@ func (mq *msgQueue) runQueue(ctx context.Context) {
 }
 
 func (pm *WantManager) Connected(p peer.ID) {
-	pm.connect <- p
+	select {
+	case pm.connect <- p:
+	case <-pm.ctx.Done():
+	}
 }
```

Comment:
I'd expect these Connected / Disconnected checks would live in a place
like this [1].  But maybe they're associated with separate
want-manager goroutine, and that's just closing a shared context on
disconnect?  Or otherwise handling the disconnect?

The consuming code is clear [2].

[1]: https://github.com/ipfs/go-ipfs/blob/41785f3b590156e05d01500d0c072b516f000a5a/exchange/bitswap/bitswap.go#L164-L177
[2]: https://github.com/ipfs/go-ipfs/blob/41785f3b590156e05d01500d0c072b516f000a5a/exchange/bitswap/wantmanager.go#L226-L229